composer/trainer/trainer.py

```diff
@@ -1316,6 +1316,20 @@ def _train_microbatch(self, use_grad_scaling: bool, current_batch_size: int, tot
 
         if self.deepspeed_enabled:
             self.state.deepspeed_model.step()
+            
+    def predict(self, loader, batchsize):
+
+        self.state.model.eval()
+        for batch_idx, self.state.batch in enumerate(
+            itertools.islice(loader, batchsize)):
+
+            self.state.batch = self._device.batch_to_device(self.state.batch)
+            if self.deepspeed_enabled:
+                self.state.batch = _fix_batch_precision_for_deepspeed(self.state.batch, self.state.precision)
+
+            with torch.no_grad():
+                outputs, targets = self._original_model.validate(self.state.batch)
+        return outputs, targets
```

Comment:
Before returning, could you restore whether the model was in training mode? (See `eval()` for how this is captured at the start of the function, and then set again at the end.)

Reply:
please point out if I missed something!